src/components/FileManager/LinkedFilesSection.tsx

```diff
@@ -0,0 +1,134 @@
+ */
+import { NodeFile } from '@manuscripts/body-editor'
+import { skipTracking } from '@manuscripts/track-changes-plugin'
+import React, { useState } from 'react'
+
+import { useStore } from '../../store'
+import { FileActions } from './FileActions'
+import { FileContainer } from './FileContainer'
+import { FileCreatedDate } from './FileCreatedDate'
+import { FileSectionType, Replace } from './FileManager'
+import { FileName } from './FileName'
+import { FileSectionAlert, FileSectionAlertType } from './FileSectionAlert'
+import { ToggleHeader } from './ToggleHeader'
+
+export type LinkedFilesSectionProps = {
+  linkedFiles: NodeFile[]
+}
+
+/**
+ * This component represents the section for linked supplemental files.
+ */
+export const LinkedFilesSection: React.FC<LinkedFilesSectionProps> = ({
+  linkedFiles,
+}) => {
+  const [isOpen, setIsOpen] = useState(false)
+  const toggleVisibility = () => setIsOpen((prev) => !prev)
+
+  const [{ view, fileManagement }] = useStore((s) => ({
+    view: s.view,
+    fileManagement: s.fileManagement,
+  }))
+
+  const [alert, setAlert] = useState({
+    type: FileSectionAlertType.NONE,
+    message: '',
+  })
+
+  if (!view) {
+    return null
+  }
+
+  const upload = async (file: File) => {
+    setAlert({
+      type: FileSectionAlertType.UPLOAD_IN_PROGRESS,
+      message: file.name,
+    })
+    const uploaded = await fileManagement.upload(file)
+    setAlert({
+      type: FileSectionAlertType.UPLOAD_SUCCESSFUL,
+      message: '',
+    })
+    return uploaded
+  }
+
+  const handleReplace = async (linkedFile: NodeFile, file: File) => {
+    const uploaded = await upload(file)
+    const tr = view.state.tr
+    tr.setNodeAttribute(linkedFile.pos, 'extLink', uploaded.id)
+    view.dispatch(skipTracking(tr))
+  }
+
+  const handleMoveToOtherFiles = (linkedFile: NodeFile) => {
+    const tr = view.state.tr
+    tr.setNodeAttribute(linkedFile.pos, 'extLink', '')
+    view.dispatch(skipTracking(tr))
+    setAlert({
+      type: FileSectionAlertType.MOVE_SUCCESSFUL,
+      message: FileSectionType.OtherFile,
+    })
+  }
+
+  return (
+    <>
+      <ToggleHeader
+        title="Linked Files"
+        isOpen={isOpen}
+        onToggle={toggleVisibility}
+      />
+      {isOpen && (
+        <>
+          <FileSectionAlert alert={alert} />
+          {linkedFiles.map((file) => (
+            <LinkedFile
+              key={file.node.attrs.id}
+              linkedFile={file}
+              onDownload={() => fileManagement.download(file.file)}
+              onReplace={(f) => handleReplace(file, f)}
+              onDetach={() => handleMoveToOtherFiles(file)}
+            />
+          ))}
+        </>
+      )}
+    </>
+  )
+}
+
+const LinkedFile: React.FC<{
+  linkedFile: NodeFile
+  onDownload: () => void
+  onReplace: Replace
+  onDetach: () => void
+}> = ({ linkedFile, onDownload, onReplace, onDetach }) => {
+  return (
+    <FileContainer data-cy="file-container" key={linkedFile.file.id}>
```

Comment:
do you need a key here? there is no loop in this component. Probably you need that only where you invoke it.